src/parser.js

```diff
@@ -31,9 +32,23 @@ export function getSliceRange(label) {
     return res ? res.slice(1) : [];
 }
 
+export function lookupLanguageByExtension(ext) {
+  for (var key in language_map) {
+    if (language_map.hasOwnProperty(key) && language_map[key].extensions) {
+      if(language_map[key].extensions.indexOf(ext) !== -1) {
+        return language_map[key].aceMode;
+      }
+    }
+  }
+
+  return undefined;
+}
+
 export function getLang(filePath) {
     var suffix = path.extname(filePath);
-    return suffix.substring(1);
+    var ext = '.' + suffix.substring(1);
```

Comment:
Is this same with that?
```js
var ext = path.extname(filePath)
```

- https://nodejs.org/api/path.html#path_path_extname_p